.github/workflows/guix.yml

```diff
@@ -0,0 +1,63 @@
+name: ci/gh-actions/guix
+
+on: [push, pull_request]
```

Comment:
I removed guix and depends package caching from the CI workflow because it uses too much space and causes evictions for other caches (10 GB limit). We only need to run it when there is a change to `contrib/{depends,guix}`, which happens infrequently and doesn't need to complete quickly.

`on: [push, pull_request]` is temporary for testing.